### DISCUSSION TITLE
Parte 3 "Killed"

### DISCUSSION THREAD

**Discussion by balliende** [IIC2133-PUC/2024-1#189] [Tarea 3] [answered]
Hola!
En la parte 3, los test 1 y 2 me corren perfecto, pero a partir del 3, el código deja de correr y me entrega como output "Killed" (debajo del limite de los 10 segundos). ¿Me podrían dar consejos de como puedo optimizar mi código?

**Comment by benjavicente** [comment]
Hola! Al correrlo con valgrind, no te sale la razón?

**Reply by balliende** [reply]
No. Solamente Killed.

**Reply by 4n1t4** [reply]
Al ver tu repo me suena extraño la forma en que generas las relaciones  al pedir un calloc de cantidad de n nodos , esto para cada nodo. Por ende, dado q la tarea es identificar grafos conexos es que no aprovecharás toda la memoria que pediste en ese calloc. Intenta representar de otra manera las relaciones de forma q solamente trabajes con la memoria necesaria.

**Reply by balliende** [reply]
Gracias, ahora si me funciona. Tenía una duda con respecto a los outputs. ¿Se les debe poner el salto de línea (\n)?

**Reply by 4n1t4** [reply]
Si te refieres al salto de línea final puede no ser necesario dado q diff-b al comparar no revisa el salto de línea final.

**Comment by 4n1t4** [comment]
Hola, q método usas para la representación de grafos, lista o matriz? Lo digo dado q podría ser un problema de memoria .

**Reply by DiegoVarela11** [reply]
Hola!
Yo presento el mismo error, pero únicamente en el test-3. Estoy utilizando matriz para representar el grafo.

**Reply by 4n1t4** [reply, marked as answer]
Intenta cambiar la forma en la cual representas , dado q en matriz probablemente haces un calloc o malloc de nxn siendo n la cantidad de nodos. Sin embargo no aprovechas todas las celdas dado que no es grafo completo. De esta forma puede ser mucho más conveniente explorar la otra forma de representación usando lista ligadas para relaciones de cada nodo

**Reply by DiegoVarela11** [reply]
yap, gracias!
